packages/core/src/app/checkout/AnalyticsEvents.ts

```diff
@@ -21,34 +19,79 @@ export enum GuestCheckoutEvents {
   Exit = "Exit"
 }
 
+const defaultTimeout = 1000;
+const defaultPollingInterval = 10;
+// borrowed polling mechanism 
+async function pollUntilAccepted<T>(
+  fn: () => T,
+  accept: (t: T) => boolean,
+  timeoutMillis = defaultTimeout,
+  interval = defaultPollingInterval,
+): Promise<T | Error> {
+  return new Promise<T | Error>(resolve => {
+    let remainingTime = timeoutMillis;
+
+    const intervalId = setInterval(() => {
+      if (remainingTime < 0) {
+        clearInterval(intervalId);
+        resolve(new Error(`Timed out while waiting for ${fn} (${timeoutMillis}ms)`));
+        return;
+      }
+
+      const value = fn();
+      if (accept(value)) {
```

Comment:
thoughts on combining accept and fn such that it polls for `timeoutMillis` milliseconds or until fn returns a non-null value? There's only a single use case right now and starting out simpler is less risky

Reply:
yeah, that's fair. I was initially just wanting to make sure the borrowed code worked in this case first

Reply:
just realized that `value` is used later in this block to resolve, so I'm leaving this as is